src/libvx.c

```diff
@@ -1153,30 +1153,28 @@ vx_error vx_frame_transfer_video_data(const vx_video* video, AVFrame* av_frame,
 	AVFrame* sw_frame = NULL;
 
 	// Copy the frame from GPU memory if it has been hardware decoded
-	bool hw_decoded = av_frame->hw_frames_ctx;
-	if (hw_decoded)
+	if (av_frame->hw_frames_ctx)
 	{
 		sw_frame = av_frame_alloc();
 
 		if (!sw_frame) {
 			result = VX_ERR_ALLOCATE;
-			goto end;
+			goto cleanup;
 		}
 
-		if (av_hwframe_transfer_data(sw_frame, av_frame, 0) < 0)
+		if (av_hwframe_transfer_data(sw_frame, av_frame, 0) < 0 | av_frame_copy_props(sw_frame, av_frame) < 0)
 		{
 			av_log(NULL, AV_LOG_ERROR, "Error transferring frame data to system memory\n");
-			av_frame_unref(sw_frame);
-			av_frame_free(&sw_frame);
-
-			goto end;
+			goto cleanup;
 		}
 
-		av_frame = sw_frame;
```

Comment:
This only updated the reference locally, so when the frame reference was used elsewhere it would be missing the updated data.